src/runtime_src/core/common/smi.h

```diff
@@ -148,6 +152,50 @@ class smi {
 
 };
 
+class config_generator {
+public:
+  virtual ~config_generator() = default;
+
+  XRT_CORE_COMMON_EXPORT
+  virtual subcommand 
+  create_validate_subcommand() = 0;
+
+  XRT_CORE_COMMON_EXPORT
+  virtual subcommand 
+  create_examine_subcommand() = 0;
+
+  XRT_CORE_COMMON_EXPORT
+  virtual subcommand 
+  create_configure_subcommand() = 0; 
```

Comment:
No symbols to export, please remove XRT_CORE_COMMON_EXPORT.
Comment each function